includes/gateway/class-omise-payment-billpayment-tesco.php

```diff
@@ -137,84 +137,92 @@ public function display_barcode( $order, $context = 'view' ) {
-							</div>
-						<?php endif; ?>
-
-						<div class="omise-billpayment-tesco-print-order-overview-item">
-							<?php _e( 'Total:', 'woocommerce' ); ?>:
+						<div>
+							<?php _e( 'Total:', 'woocommerce' ); ?>
 							<br/><strong><?php echo $this->order()->get_formatted_order_total(); ?></strong>
 						</div>
-					</div>
 
-					<?php if ( $this->order()->get_payment_method_title() ) : ?>
-						<div class="omise-billpayment-tesco-print-order-payment-method">
-							<?php _e( 'Payment method:', 'woocommerce' ); ?>
-							<br/><strong><?php echo wp_kses_post( $this->order()->get_payment_method_title() ); ?></strong>
-						</div>
-					<?php endif; ?>
+						<?php if ( $this->order()->get_payment_method_title() ) : ?>
+							<div>
+								<?php _e( 'Payment method:', 'woocommerce' ); ?>
+								<br/><strong><?php echo wp_kses_post( $this->order()->get_payment_method_title() ); ?></strong>
+							</div>
+						<?php endif; ?>
+					</div>
+				<?php endif; ?>
+				<!-- /END .omise-billpayment-tesco-print-detail -->
+
+				<p>
+					<?php
+					echo sprintf(
+						wp_kses(
+							__( 'Please bring this barcode to pay at Tesco Lotus within:<br/><strong>%1$s</strong> at <strong>%2$s</strong>.', 'omise' ),
+							array( 'br' => array(), 'strong' => array() )
+						),
+						wc_format_datetime( $expires_datetime, wc_date_format() ),
+						wc_format_datetime( $expires_datetime, wc_time_format() )
+					);
+					?>
+				</p>
+
+				<div class="omise-billpayment-tesco-barcode-wrapper" <?php echo 'email' === $context ? 'style="margin: 1em 0;"' : ''; ?>>
+					<?php echo $barcode_html; ?>
+				</div>
 
-					<p class="omise-billpayment-tesco-print-barcode-message"><?php _e( 'Use this barcode to pay at Tesco Lotus.', 'omise' ); ?></p>
+				<div class="omise-billpayment-tesco-barcode-reference-number">
+					<small><?php echo $barcode_ref_number; ?></small>
+				</div>
 
-					<div class="omise-billpayment-tesco-print-barcode-wrapper">
-						<?php echo $barcode_html; ?>
+				<?php if ( 'email' !== $context ) : ?>
+					<div class="omise-billpayment-tesco-print-button">
+						<button onClick="window.print()" class="button button-primary">Print barcode</button>
 					</div>
+				<?php endif; ?>
 
-					<div class="omise-billpayment-tesco-print-reference-number">
-						<small><?php echo $barcode_ref_number; ?></small>
-					</div>
+				<div class="omise-billpayment-tesco-footnote" <?php echo 'email' === $context ? 'style="margin-top: 4em; padding: 1em; font-size: 85%; text-align: right; background-color: #f8f8f8;"' : ''; ?>>
+					<p <?php echo 'email' === $context ? 'style="margin: 0; padding: 0;"' : ''; ?>>
+						<?php
+						echo wp_kses(
+							__(
+								'
+								Tesco Lotus may charge a small fee for the transaction.<br/>
+								Failing to make payment within the mentioned date and time, your order will be automatically canceled.
+								', 'omise'
```

Comment:
`'If you fail to make payment by the stated time, your order will be automatically canceled.'`

Or `cancelled` if we're going for British English :P

Reply:
@guzzilar are we sure we have 24 hours to make a payment, not 48 hours?

Reply:
😂 Btw, @jonrandy  wonder why `If you fail to make payment` is better than `Failing to make payment`?

Reply:
Or if we want to make it more polite, how about `Please note that failing to make payment...`? or it is not about politeness?

Reply:
If you want to use 'failing to make payment', you need some further rewording...
`Please note that failing to make payment by the stated time will result in your order being automatically canceled.`
Which is getting a little bit long winded